File: frontend/src/types/health.ts
export type ClusterHealth = {
	summary: HealthSummary;
	nodes: Record<number, NodeHealth>;
};

export type HealthSummary = {
	online: number;
	total: number;
};

export type NodeHealth = {
	id: number;
	name: string;
	status: NodeStatus;
	latencyMs: number;
	lastErr?: string;
	piholeVersion?: string;
	ftlVersion?: string;
	gravityCount?: number;
	gravityUpdatedAt?: number; // unix timestamp (seconds)
};

export const NodeStatus = {
	ONLINE: 'online',
	OFFLINE: 'offline',
	DEGRADED: 'degraded',
} as const;
export type NodeStatus = (typeof NodeStatus)[keyof typeof NodeStatus];

File: backend/internal/domain/health.go
package domain

import "time"

type ClusterHealth struct {
	Summary   ClusterHealthSummary
	Nodes     map[int64]ClusterNodeHealth
	UpdatedAt time.Time
}

type ClusterHealthSummary struct {
	Online int
	Total  int
}

type ClusterNodeHealth struct {
	Id          int64
	Name        string
	Status      ClusterHealthStatus
	Latency     time.Duration
	LastErr     string
	VersionInfo *NodeVersionInfo
	DBInfo      *NodeDBInfo
}

type NodeVersionInfo struct {
	PiholeVersion string
	FTLVersion    string
}

type NodeDBInfo struct {
	GravityCount     int64
	GravityUpdatedAt *time.Time
}

type ClusterHealthStatus string

const (
	ClusterHealthStatusOnline   ClusterHealthStatus = "online"
	ClusterHealthStatusDegraded ClusterHealthStatus = "degraded"
	ClusterHealthStatusOffline  ClusterHealthStatus = "offline"
)
